lib/server.go

```diff
@@ -220,6 +219,9 @@ func (srv *Server) GetBlockchain() *Blockchain {
 
 // TODO: The hallmark of a messy non-law-of-demeter-following interface...
 func (srv *Server) GetMempool() Mempool {
+	srv.blockchain.ChainLock.RLock()
+	defer srv.blockchain.ChainLock.RUnlock()
```

Comment:
Out of curiosity, why do we need to acquire this lock here? Is it just to make sure the tip height doesn't shift underneath us?